src/openfl/display/DisplayObjectContainer.hx

```diff
@@ -489,18 +489,23 @@ class DisplayObjectContainer extends InteractiveObject {
 		if (__cacheBitmap != null && !__cacheBitmapRender)
 			return;
 
-		renderSession.maskManager.pushObject(this);
+		var needRender = renderSession.maskManager.pushObject(this);
+		if (needRender) {
+			if (renderSession.clearRenderDirty) {
+				for (child in __children) {
+					child.__renderCanvas(renderSession);
+					child.__renderDirty = false;
+				}
 
-		if (renderSession.clearRenderDirty) {
-			for (child in __children) {
-				child.__renderCanvas(renderSession);
-				child.__renderDirty = false;
+				__renderDirty = false;
+			} else {
+				for (child in __children) {
+					child.__renderCanvas(renderSession);
+				}
 			}
-
-			__renderDirty = false;
-		} else {
+		} else if (renderSession.clearRenderDirty) {
 			for (child in __children) {
-				child.__renderCanvas(renderSession);
+				child.__renderDirty = false;
 			}
```

Comment:
Don't we need a `__renderDirty = false;` like in line 500?

Reply:
Good catch!